### ISSUE TITLE
`LightCurveFile.header` is deprecated when downloading data.

### ISSUE THREAD

**Issue by warrickball** [KeplerGO/lightkurve#800]
#### Problem description

Downloading data emits the warning

    LightkurveWarning: `LightCurveFile.header` is deprecated, please use `LightCurveFile.get_header()` instead.

even though I'm not using `LightCurveFile.header` at all.

#### Example

```python
import lightkurve
lcs = lightkurve.search_lightcurvefile('rho pup', mission='tess', cadence='short').download_all()
```
The warning only comes from the `download` commands. This emits no warning
```python
import lightkurve
lcs = lightkurve.search_lightcurvefile('rho pup', mission='tess', cadence='short') # no warning
```
but this does
```python
import lightkurve
lcs = lightkurve.search_lightcurvefile('rho pup', mission='tess', cadence='short')[0].download()
```

#### Expected behavior

The data is downloaded (which it is) without a warning.

#### Environment

* Fedora 32
* Python 3.8.5, provided by the OS.
* lightkurve 1.11.1, installed with `pip3 install lightkurve --user`. I tried upgrading and this appears to be the latest version available to me.

**Comment by ojhall94** [comment]
Hey @warrickball!

I've just tested this out on the latest GitHub version of Lightkurve, and the issue is no longer present. There were some calls to deprecated functions left over when we shuffled around the `header` and `LightCurveFile` system, which I think should all be resolved now. I'm not sure if @barentsen wants to push a minor update to fix this in the current release, but it will definitely be gone by the next update :)

**Comment by barentsen** [comment]
@warrickball  Thank you for drawing attention to this annoying warning :+1:

I fixed the warning and released v1.11.2 on PyPI, i.e. `python -m pip install lightkurve --upgrade` should remove this warning!

The warning does not affect the current master branch i.e. Lightkurve v2.0.dev.

Thanks again!

**Comment by warrickball** [comment]
I just upgraded to 1.11.2 and can confirm that the warning is gone. Thanks!
